### DISCUSSION TITLE
setAuthCookies doesn't set cookies after login/logout/login and doesn't throw errors when using withAuthUserTokenSSR

### DISCUSSION THREAD

**Discussion by Felixaverlant** [gladly-team/next-firebase-auth#538] [Q&A]
Hi,

I'm currently using `next-firebase-auth ^1.0.0-canary.15` and I have a problem setting and unsetting the cookies ( I think).

Here's the steps : 
- launching dev server
- hit login => get correctly logged in, get redirect to app.
- signout (through `AuthUser.signOut`), get redirect to login page
- hit login => /api/login http request keeps being pending.
- relaunch dev server => Works again.

Notes
- The pending state is during  `await setAuthCookies(req,res)`
- In the front log, I can see that AuthUser returns the correct user but I can't visit a withAuthUser page.
- I had to had `return res.status(200).json({ status: true })` in both /api/login and /api/logout because otherwise the pending http request happens even when I login for the first time after launch dev server. 
- it only occurs when I use `withAuthUserTokenSSR` but works with only `withAuthUser`
Here's my current setup : 

**login.ts**
```js
import { NextApiRequest, NextApiResponse } from 'next'
import { setAuthCookies } from 'next-firebase-auth'
import initAuth from '../../lib/initAuth'

initAuth()

const handler = async (req: NextApiRequest, res: NextApiResponse) => {
  try {
    await setAuthCookies(req, res)
    // If I don't have this return, the pending state happens even when I restart
    return res.status(200).json({ status: true })
  } catch (e) {
    console.error(e)
    return res.status(500).json({ error: 'Unexpected error.' })
  }
  return res.status(200).json({ status: true })
}

export default handler
```

**logout.ts**
```js
initAuth()

const handler = async (req, res) => {
  try {
    await unsetAuthCookies(req,res)
    // If I don't have this return, the pending state happens even when I restart
    return res.status(200).json({ status: true })
  } catch (e) {
    console.error(e)
    return res.status(500).json({ error: 'Unexpected error.' })
  }
}
```

**Comment by kmjennison** [comment]
The behavior you're describing makes sense to me: the server needs to send a response, otherwise the request will hang. Unless I'm misunderstanding, this doesn't seem like a problem or anything necessarily tied to `next-firebase-auth`.
